optimum/exporters/ipex/cache_utils.py

```diff
@@ -43,10 +45,14 @@ def __init__(
     ) -> None:
         super().__init__()
         self.max_batch_size = max_batch_size
+        self.device = device
+        self.flash_decoding = (
```

Comment:
let's use `_supports_flash_decoding`, also it makes sense to add it as a private variable since the api seems to change a lot.